blocks/snippets.py

```diff
@@ -0,0 +1,14 @@
+"""Routines to build frequently used Theano expressions."""
+from theano import tensor
+
+
+def L2_norm(tensors):
+    """Computes the total L2 norm of a set of tensors.
+
+    Parameters
+    ----------
+    tensors : iterable of :class:`~theano.Variable`
+        The tensors.
+
+    """
+    return tensor.sqrt(sum((t ** 2).sum() for t in tensors))
```

Comment:
Bricks are also not like bricks at all, they have zero clay in them!

For me the difference between calling my plugin to substitute inputs in a small piece of code or calling a function to do that is minor. I like to call things according to their purpose. There is a need for a file with small pieces of Theano code: I call it snippets, never caring whether the code is actually copy-pasted or not.

I would be opposed to replacing `expressions` with `variables` in many cases in the code, so let's keep it slow.

`theano_utils` has a disadvantage that for many functions from `utils.py` it is very hard to say where they should belong, to `theano_utils.py` or to `utils.py`. I am fine with "theano_macros¨, though your arguments apply fully to this name as well.